packages/flutter_tools/test/general.shard/fake_process_manager_test.dart

```diff
@@ -172,4 +175,149 @@ void main() {
+
+        final ProcessResult result = await manager.run(<String>['faketool']);
+        expect(result.exitCode, 0);
+        expect(result.stdout, isEmpty);
+        expect(result.stderr, isEmpty);
+      });
+
+      testWithoutContext('stderr, stdout are String if encoding is unspecified', () async {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final ProcessResult result = await manager.run(<String>['faketool']);
+        expect(result.exitCode, 0);
+        expect(result.stdout, isA<String>());
+        expect(result.stderr, isA<String>());
+      });
+
+      testWithoutContext('stderr, stdout are List<int> if encoding is null', () async {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final ProcessResult result = await manager.run(
+          <String>['faketool'],
+          stderrEncoding: null,
+          stdoutEncoding: null,
+        );
+        expect(result.exitCode, 0);
+        expect(result.stdout, isA<List<int>>());
+        expect(result.stderr, isA<List<int>>());
+      });
+
+      testWithoutContext('stderr, stdout are List<int> if encoding is specified', () async {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final ProcessResult result = await manager.run(
+          <String>['faketool'],
+          stderrEncoding: utf8,
+          stdoutEncoding: utf8,
+        );
+        expect(result.exitCode, 0);
+        expect(result.stdout, isA<String>());
+        expect(result.stderr, isA<String>());
+      });
+    });
+
+    group('runSync', () {
+      testWithoutContext('can run a fake command', () {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final ProcessResult result = manager.runSync(<String>['faketool']);
+        expect(result.exitCode, 0);
+        expect(result.stdout, isEmpty);
+        expect(result.stderr, isEmpty);
+      });
+
+      testWithoutContext('stderr, stdout are String if encoding is unspecified', () {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final ProcessResult result = manager.runSync(<String>['faketool']);
+        expect(result.exitCode, 0);
+        expect(result.stdout, isA<String>());
+        expect(result.stderr, isA<String>());
+      });
+
+      testWithoutContext('stderr, stdout are List<int> if encoding is null', () {
+        manager.addCommand(const FakeCommand(command: <String>['faketool']));
+
+        final ProcessResult result = manager.runSync(
+          <String>['faketool'],
+          stderrEncoding: null,
+          stdoutEncoding: null,
+        );
+        expect(result.exitCode, 0);
+        expect(result.stdout, isA<List<int>>());
+        expect(result.stderr, isA<List<int>>());
+      });
+
+      testWithoutContext('stderr, stdout are List<int> if encoding is specified', () {
```

Comment:
Is this test message correct? It looks like the expectation is `String` instead:

```suggestion
      testWithoutContext('stderr, stdout are String if encoding is specified', () {
```

Reply:
Also, is this test a duplicate of the test that starts on line 263?

Reply:
Thanks! A little to quick on the copy-paste there!